tensorflow_quantum/python/optimizers/rotosolve_minimizer.py

```diff
@@ -0,0 +1,284 @@
+
+    ### Usage:
+
+    The following example demonstrates the Rotosolve optimizer attempting
+     to find the minimum for two qubit ansatz expectation value.
+
+    ```python
+
+    # In this example we train a circuit perform an XOR operation
+    X = np.asarray([
+        [0, 0],
+        [0, 1],
+        [1, 0],
+        [1, 1],
+    ], dtype=float)
+
+    Y = np.asarray([
+        [-1], [1], [1], [-1]
+    ], dtype=float)
+
+    def convert_to_circuit(input_data):
+        # Encode into quantum datapoint.
+        values = np.ndarray.flatten(input_data)
+        qubits = cirq.GridQubit.rect(1, 2)
+        circuit = cirq.Circuit()
+        for i, value in enumerate(values):
+            if value:
+                circuit.append(cirq.X(qubits[i]))
+        return circuit
+
+    x_circ = tfq.convert_to_tensor([convert_to_circuit(x) for x in X])
+
+    # Create two qubits
+    q0, q1 = cirq.GridQubit.rect(1, 2)
+
+    # Create an anzatz on these qubits.
+    a, b = sympy.symbols('a b') # parameters for the circuit
+    circuit = cirq.Circuit(
+        cirq.rx(a).on(q0),
+        cirq.ry(b).on(q1), cirq.CNOT(control=q0, target=q1))
+
+    # Build the Keras model.
+    model = tf.keras.Sequential([
+        # The input is the data-circuit, encoded as a tf.string
+        tf.keras.layers.Input(shape=(), dtype=tf.string),
+        # The PQC layer returns the expected value of the
+        # readout gate, range [-1,1].
+        tfq.layers.PQC(circuit, cirq.Z(q1)),
+    ])
+
+    def hinge_loss(y_true, y_pred):
+        # Here we use hinge loss as the cost function
+        tf.reduce_mean(tf.cast(1 - y_true * y_pred, tf.float32))
+
+    result = rotosolve_minimizer.minimize(
+        rotosolve_minimizer.function_factory(
+            model, hinge_loss, x_circ, Y), np.random.rand([2])
+            ) # Initial guess of the parameter
+
+    print(result)
+    ```
+
+    Args:
+      expectation_value_function:  A Python callable that accepts
+        a point as a real `Tensor` and returns a tuple of `Tensor`s
+        of real dtype containing the value of the function and its
+        gradient at that point. The function to be minimized. The
+        input is of shape `[..., n]`, where `n` is the size of the
+        domain of input points, and all others are batching dimensions.
+        The first component of the return value is a real `Tensor` of
+        matching shape `[...]`. The second component (the gradient)
+        is also of shape `[..., n]` like the input value to the function.
+        This must be a linear combination of quantum measurement
+        expectation value, otherwise this algorithm cannot work.
+      initial_position: Real `Tensor` of shape `[..., n]`. The starting
+        point, or points when using batching dimensions, of the search
+        procedure. At these points the function value and the gradient
+         norm should be finite.
+      tolerance: Scalar `Tensor` of real dtype. Specifies the gradient

```

Comment:
`tf.Tensor`

Reply:
Fixed in 25a75df2f6c70559080182fcc5f2945585479b8d